include/boost/geometry/strategies/spherical/distance_haversine.hpp

```diff
@@ -94,8 +94,20 @@ public :
                );
     }
 
+    template <unsigned int Index, typename Point1, typename Point2>
+    inline typename calculation_type<Point1, Point2>::type
+    coordinate(Point1 const& p1, Point2 const& p2)
+    {
+        BOOST_CONCEPT_ASSERT( (concepts::ConstPoint<Point1>) );
+        BOOST_CONCEPT_ASSERT( (concepts::ConstPoint<Point2>) );
+
+        assert_dimension_equal<Point1, Point2>();
+
+        return geometry::get<Index>(p1) - geometry::get<Index>(p2);
+    }
+
     template <typename T1, typename T2>
-    inline radius_type meridian(T1 lat1, T2 lat2) const
+    inline radius_type vertical_or_meridian(T1 lat1, T2 lat2) const
```

Comment:
Same here. Is this element of the interface requried by the distance P/P algorithm? If it's a CS specific optimization used in some other strategy it should be implemented there, not here.